server/images.go

```diff
@@ -514,6 +514,14 @@ func embeddingLayers(e EmbeddingParams) ([]*LayerReader, error) {
 					continue
 				}
 				addedFiles[filePath] = true
+				// check if we already have embeddings for this file path
+				layerIdentifier := fmt.Sprintf("%s:%s", filePath, e.model)
```

Comment:
Shouldn't this be hashing the contents of the file?

Reply:
updated to use the digest of the file bin (which should change when the weights change)